doc: change mailing list references to github (#3082)
UIUC-PPL/charm: README.md

The Charm++ documentation is at https://charm.readthedocs.io/

The Charm++ web page, with more information, more programs,
and the latest version of Charm++, is at https://charmplusplus.org

The UIUC Parallel Programming Laboratory web page, with information
on past and present research, is at https://charm.cs.illinois.edu

For questions, comments, suggestions, improvements, or bug reports,

> please contact us on our GitHub, https://github.com/UIUC-PPL/charm

Reviewer: Maybe this and the other references should specifically mention Discussions. This isn't clear to me right now.

Author: My thinking was that the user should choose which "section" of the GitHub is most suitable (e.g., Discussions, Issues, PRs) for their situation, which is why I didn't specify Discussions in most places; but I'm not sure if this is the best way to do it.

Reviewer: Yeah, fair. Then maybe just say something like "create an issue or discussion on our GitHub"? There are likely users who aren't familiar with GitHub, and even for those who are, I feel like just saying GitHub is too vague.

Author: Good idea, added your suggestion to several locations in the text.

> For questions, comments, suggestions, improvements, or bug reports,
> please contact us on our GitHub, https://github.com/UIUC-PPL/charm

Author: ```suggestion
For questions, comments, suggestions, improvements, or bug reports,
please create an issue or discussion on our GitHub, https://github.com/UIUC-PPL/charm
```
